### PR TITLE
Separate userdata from storagedata, move both from auth.

### PR DESCRIPTION
**Pull request by radiansz**

### CODE REVIEW

File: client/logic/modules/auth/__tests__/reducers.js [hunk 1]
```diff
@@ -1,57 +1,23 @@
...
-  FETCH_USERDATA_REQUEST,
-  FETCH_USERDATA_SUCCESS,
-  FETCH_USERDATA_FAILURE,
   LOGOUT_REQUEST,
 } from '../actions';
-import authReducer, { token, userdata } from '../reducers';
+import authReducer from '../reducers';
 
 describe('auth', () => {
   test('should return the initial state', () => {
```
Comment (rosendi) [inline client/logic/modules/auth/__tests__/reducers.js]: `test('returns`

File: client/logic/modules/auth/__tests__/reducers.js [hunk 2]
```diff
@@ -1,57 +1,23 @@
...
-    test('should return the initial state', () => {
-      expect(token(undefined, {})).toEqual(null);
-    });
-
-    test('should handle AUTHENTICATE', () => {
-      const payload = 'token';
-
-      expect(token({}, { type: AUTHENTICATE, payload })).toEqual(payload);
-    });
+  test('should handle AUTHENTICATE', () => {
```
Comment (rosendi) [inline client/logic/modules/auth/__tests__/reducers.js]: `test('handles`

File: client/logic/modules/auth/__tests__/reducers.js [hunk 3]
```diff
@@ -1,57 +1,23 @@
...
-    test('should handle FETCH_USERDATA_SUCCESS', () => {
-      const payload = { foo: 'bar' };
-
-      expect(userdata({}, { type: FETCH_USERDATA_SUCCESS, payload })).toEqual({ data: payload, fetching: false });
-    });
-
-    test('should handle FETCH_USERDATA_FAILURE', () => {
-      expect(userdata({ fetching: true }, { type: FETCH_USERDATA_FAILURE })).toEqual({ fetching: false });
-    });
+  test('should handle LOGOUT_REQUEST', () => {
```
Comment (rosendi) [inline client/logic/modules/auth/__tests__/reducers.js]: `test('handles`

File: client/logic/modules/auth/reducers.js [hunk 4]
```diff
@@ -16,23 +16,4 @@ export const token = (state = null, { type, payload }) => {
...
-      return { ...state, fetching: false };
-    }
-
-    default:
-      return state;
-  }
-};
-
-export default combineReducers({ token, userdata });
+export default combineReducers({ token });
```
Comment (rosendi) [inline client/logic/modules/auth/reducers.js]: It isn't necessary to combine a single reducer. Keep it flatten. Just `export default token`.

File: client/logic/modules/changelog/__tests__/reducers.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  FETCH_CHANGELOG_REQUEST,
+  FETCH_CHANGELOG_SUCCESS,
+  FETCH_CHANGELOG_FAILURE,
+  SAVE_CHANGELOG_REQUEST,
+  SAVE_CHANGELOG_SUCCESS,
+  SAVE_CHANGELOG_FAILURE,
+} from '../actions';
+import changelogReducer from '../reducers';
+
+describe('changelog', () => {
```
Comment (rosendi) [inline client/logic/modules/changelog/__tests__/reducers.js]: Please remove the `should` prefix and append the first verbs with the `s` suffix.

File: client/logic/modules/changelog/index.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import reducer from './reducers';
+
+export { fetchChangelog, saveChangelog } from './creators';
+export { default as saga } from './sagas';
+export default reducer;
```
Comment (rosendi) [inline client/logic/modules/changelog/index.js]: export { default } reducer;

File: client/logic/modules/changelog/index.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import reducer from './reducers';
```
Comment (rosendi) [inline client/logic/modules/changelog/index.js]: remove

File: client/logic/modules/changelog/sagas.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        lastSeenAt = storage.obj.data.lastSeenAt;
+      }
+
+      yield put(fetchChangelogDone({ lastSeenAt }));
+    } catch (err) {
+      yield put(fetchChangelogError(err));
+    }
+  }
+}
+
```
Comment (rosendi) [inline client/logic/modules/changelog/sagas.js]: May be just:

```
export default function* root() {
  yield [
    fork(changelogSave),
    fork(changelogFetch),
  ]
}
```

File: client/logic/modules/changelog/reducers.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { FETCH_CHANGELOG_REQUEST, FETCH_CHANGELOG_SUCCESS, FETCH_CHANGELOG_FAILURE,
```
Comment (rosendi) [inline client/logic/modules/changelog/reducers.js]: Turn each constant into a new line.

**Comment by radiansz** [comment]
Fixed

**Comment by zouxuoz** [comment]
LGTM

**Comment by rosendi** [comment]
lgtm

**Pull request merged**